### ISSUE TITLE
Swiper changes the faces of text buttons

### ISSUE THREAD

**Issue by Alexander-Miller** [abo-abo/swiper#1212]
Steps to reproduce:

 * emacs -q
 * load swiper
 * switch to a text-mode buffer
 * `(insert-text-button "AAAAAAAAA" 'face 'font-lock-constant-face)`
 * Run swiper
 * The inserted button's face has now changed from constant to button

Strangely enough this only happens once. After the first face change this behaviour stops. 
I tried both emacs 25 and emacs 26, same as my other issue.

**Comment by abo-abo** [comment]
Is this because of `font-lock-ensure`? In that case, it's an Emacs bug.

**Comment by Alexander-Miller** [comment]
Looks like it is. I've opened [a bug ticket](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=28599) for it.

Since this is not swiper-specific I'll close this issue.

**Comment by abo-abo** [comment]
Thanks, closing.

**Comment by abo-abo** [comment]
If you want, I'll add an offending major-mode to `swiper-font-lock-exclude` - a quick temporary fix.

**Comment by Alexander-Miller** [comment]
> Thanks, closing.

Could've sworn I did that already. At least I wanted to.

> swiper-font-lock-exclude

I think I ran into one of your blog posts about that when looking up font-lock-ensure - needing to make sure that the entire buffer is fontified for swiper? If so the exclusion is a good idea, other than assigning buttons' faces treemacs has no interaction with font locking. The major-mode's symbol is simply `treemacs-mode`.
